server/src/internalClusterTest/java/org/elasticsearch/indices/IndexingMemoryControllerIT.java

```diff
@@ -111,4 +114,21 @@ public void testDeletesAloneCanTriggerRefresh() throws Exception {
         }
         assertThat(shard.getEngineOrNull().getIndexBufferRAMBytesUsed(), lessThanOrEqualTo(ByteSizeUnit.KB.toBytes(1)));
     }
+
+    /* When there is memory pressure, we write indexing buffers to disk on the same thread as the indexing thread,
+     * @see org.elasticsearch.indices.IndexingMemoryController.
+     * This test verifies that we update the stats that capture the combined time for indexing + writing the
+     * indexing buffers.
+     */
+    public void testIndexingUpdatesRelevantStats() throws Exception {
+        IndexService indexService = createIndex("index", indexSettings(1, 0).put("index.refresh_interval", -1).build());
+        IndexShard shard = indexService.getShard(0);
+        for (int i = 0; i < 100; i++) {
+            prepareIndex("index").setId(Integer.toString(i)).setSource("field", "value").get();
```

Comment:
can we use a random value? I mention this because using the same literal value will make the flush/refresh operation quite fast (there's just a single compressed value to write) and it might make the test flaky.